content/java/concepts/hash-map/hash-map.md

```diff
@@ -0,0 +1,217 @@
+---
+Title: 'HashMap'
+Description: 'In Java, a HashMap is used to store items as a "key/value" pairs. The keys and values can be of either same or different types. java // Create a HashMap which stores courses and their teachers as a key/value pair  HashMap<String, String> courseTeacher = new HashMap<String, String>();'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'HashMap'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+In Java, a HashMap is used to store items as key-value pairs. The key and value can be of the same or different data type.
+
+```java
+// Create a HashMap which stores courses and their teachers as a key/value pair
+HashMap<String, String> courseTeacher = new HashMap<String, String>();
+```
+
+## Creating a HashMap
+
+In Java, a `HashMap` can be created in the following way:
+
+````java
+HashMap<KeyDatatype, ValueDatatype> HashMapName = new HashMap<KeyDatatype, ValueDatatype>();
+
+```java
+// Import the HashMap class
+import java.util.HashMap;
+
+  // Create a HashMap which stores courses and their teachers as a key/value pair
+HashMap<String, String> courseTeacher = new HashMap<String, String>();
+````
+
+Here, `courseTeacher` will store keys of courses with `String` type and values of Teacher assigned for the course also of `String` type.
+
+```java
+// Import the HashMap class
+import java.util.HashMap;
+
+// Create a HashMap which stores students and their marks as a key-value pair
+HashMap<String, Integer> studentMarks = new HashMap<String, Integer>();
+```
+
+Here, `studentMarks` stores keys of student names of `String` type and values of marks secured by the student of `Integer` type.
+
+## Key-Value
+
+A key can be used to uniquely identify a value in a `HashMap`. This allows for efficient data storage and easy access. In the example below, the course names are the keys and the teachers assigned are the values that can be accessed using the corresponding key.
+
+```java
+// Import the HashMap class
+import java.util.HashMap;
+
+public class Main {
+  public static void main(String[] args) {
+
+    HashMap<String, String> courseTeacher = new HashMap<String, String>();
+
+    // Add keys and values (CourseNames, Teacher)
+    courseTeacher.put("History", "Ben");
+    courseTeacher.put("Mathematics", "Jeanette");
+    courseTeacher.put("Physics", "Lily");
+
+    System.out.println(courseTeacher.get("Physics"););
+    // Output: Lily
+
+    System.out.println(courseTeacher.get("History"););
```

Comment:
For Docs, it isn't best practice to use techniques/functions/methods/etc. before they are formally introduced. This includes:

- Using `.put()` before it is introduced in "## Adding Items".
- Using `.get()` before it is introduced in "## Accessing an Item".